src/proposer/proposer.cpp

```diff
@@ -29,15 +30,83 @@
 
 namespace proposer {
 
-class ProposerStub : public Proposer {
+bool GenerateBlock(staking::ActiveChain &active_chain,
```

Comment:
According to logs, this function could be renamed to `TryPropose`, which makes sort of collision with ProposerLogic, but represents function nature much better.

Reply:
I don't agree, `TryPropose` at the moment is about checking if the coin wins the lottery, here we are actually building a block.